docs/API/V1/contents.md

```diff
@@ -88,11 +91,44 @@ Note that symlinks within the dataset representation will result in a URI return
 }
 ```
 
-### File
+#### Subirectory object
+
+The subdirectory object gives the name of the directory, the type of the entry,
+and a URI that can be used with a subsequent `GET` operation to return a
+[directory object](#directory-object) for that nested path.
 
-When the specified path is a leaf file (regular file or symlink), the response body is an `application/json` formatted [file object](#file-object), including the name and filesystem metadata. A URI is provided which can be used to `GET` the file's contents as a raw byte stream.
+When a directory contains a symlink to a directory, that subdirectory name will
+appear in the `directories` list, but will be designated with a `type` of
+`sym` instead of `dir`.
 
-#### Example
+The `type` codes are:
+* `dir`: Directory
+* `sym`: Symbolic link
+
+```json
+{
+    "name": "reference-result",
+    "type": "sym",
+    "uri": "http://hostname/api/v1/datasets/contents/<id>/sample1"
+}
+```
+
+#### File object
+
+The file object gives the name of the file, file system information about that
+file, and a URI that can be used with a subsequent `GET` operation to return
+the raw byte stream of that file.
+
+The file system information includes:
+* `mtime`: the file's last modification time,
+* `size`: the size of the file,
+* `mode`: the file permissions (as an octal "mode" string), and
+* `type`: the file's type. The type values are:
+  * `reg`: Regular UNIX file
+  * `sym`: Symbolic link
+
+Note that symlinks to files within the dataset representation will result in a
+URI returning the linked file's byte stream.
```

Comment:
If the file is a symlink, do we get `mtime`, `size`, and `mode`, and, if so, do they refer to the symlink file itself or to its referent?  (I thought we were going to omit those three fields for symlinks.)

Reply:
I'm thinking, broadly, of my experiences with symlinks in the filesystem.  Generally speaking, when accessing a symlink, tools instead access the file system object to which it refers.  This is obviously not true of `ls`, but to my surprise it was true of `chown` (unless you include the `-h` switch).

And, per the `chmod` man page:
> `chmod` never changes the permissions of symbolic links; the `chmod` system call cannot change their permissions.  This is not a problem since the permissions of symbolic links are never used.  However, for each symbolic link listed on the command line, `chmod` changes the permissions of the pointed-to file.

So, I'm (generally) thinking that the `mtime`, `size`, and `mode` of the symlink file itself aren't worth bothering about.  (Obviously, the `mode` is useless -- `ls` reports it as `lrwxrwxrwx` -- and, as you said, the `size` is of limited use; the `mtime` could conceivably be significant, but it's probably not as important as the `mtime` of the file-object to which the link refers.)

So, I would be inclined to do whatever is easiest for symlinks; and, I suspect that that is to omit the three fields altogether!  And, I would do so for both file objects and directory objects....